plugins/block-dynamic-connection/src/dynamic_if.ts

```diff
@@ -11,25 +11,20 @@
 
 import * as Blockly from 'blockly/core';
 
+type DynamicIfBlock = Blockly.Block | (typeof Blockly.Blocks)['dynamic_if'];
```

Comment:
I don't think it's strictly correct: really it should be an intersection, not a union:
```suggestion
type DynamicIfBlock = Blockly.Block & typeof Blockly.Blocks['dynamic_if'];
```
but I think this is still not correct: the type the values of the `Blockly.Blocks` dictionary is `BlockDefinition`, which is just an alias for `any`, so this line just means `type DynamicIfBlock = any`.

This would be why `tsc` is not complaining about (e.g.) your calling `this.getInput` despite `.getInput` not existing on all members of the union (in particular, `Blockly.Blocks['dynamic_if']` does not declare a `.getInput` property).

Unfortunately when I tried getting this right in core (in `blocks/*.ts`) I had to split the code defining the block out from the code installation in the block dictionary, and using [this ~~triple-headed monster~~ three-line monstrosity](https://github.com/google/blockly/blob/5f2e143ab8821ad54331cbba231d31a834ea5c8f/blocks/math.ts#L433) to work around errors due to (alleged) circular definitions.

Reply:
Done, I imitated your 3 line monstrosity, thanks!